core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -354,8 +356,45 @@ public Response getBucket(@PathParam("bucket") final String bucket,
             if (prefixParam != null) {
               path = parsePathWithDelimiter(path, prefixParam, AlluxioURI.SEPARATOR);
             }
-            ListStatusPOptions options = ListStatusPOptions.newBuilder().setRecursive(true).build();
-            children = userFs.listStatus(new AlluxioURI(path), options);
+            String startAfter = (markerParam != null ? markerParam
+                : (startAfterParam != null ? startAfterParam : ""));
+            ListStatusPartialPOptions options = ListStatusPartialPOptions.newBuilder()
+                .setStartAfter(startAfter)
+                .setBatchSize(maxKeys)
```

Comment:
can i ask what's the startAfter could be? will it be an actual obj full path or will it be some opaque value to start listing after that? and will startAfter and prefix be given at the same time? and what's the use scenario here? ( basically will there be a case like:
1) prefix  a/c/d   startafter  a/b ?
2) startafter  a   (opaque val)
3) prefix a/b  startafter a/c ?
or could u give us info on how these flags have been used by upper application?

Reply:
When the result is not fully returned, use the marker or continuation-token of the next request as the startAfter of listStatusPartial. It is a complete path. I found some problems with the startAfter assignment, which has been fixed.